lxd/deployments.go

```diff
@@ -0,0 +1,2067 @@
+//    "403":
+//      $ref: "#/responses/Forbidden"
+//    "500":
+//      $ref: "#/responses/InternalServerError"
+
+// swagger:operation GET /1.0/deployments/{deploymentName}/shapes?recursion=1 deployments deployment_shapes_get_recursion1
+//
+//	Get the deployment deploymentShapes
+//
+//	Returns a list of deployment deploymentShapes (structs).
+//
+//	---
+//	produces:
+//	  - application/json
+//	parameters:
+//	  - in: query
+//	    name: project
+//	    description: Project name
+//	    type: string
+//	    example: default
+//	responses:
+//	  "200":
+//	    description: API endpoints
+//	    schema:
+//	      type: object
+//	      description: Sync response
+//	      properties:
+//	        type:
+//	          type: string
+//	          description: Response type
+//	          example: sync
+//	        status:
+//	          type: string
+//	          description: Status description
+//	          example: Success
+//	        status_code:
+//	          type: integer
+//	          description: Status code
+//	          example: 200
+//	        metadata:
+//	          type: array
+//	          description: List of deployment shapes
+//	          items:
+//	            $ref: "#/definitions/DeploymentShape"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func deploymentShapesGet(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	projectName := projectParam(r)
+
+	deploymentName, err := url.PathUnescape(mux.Vars(r)["deploymentName"])
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	var dbDeployment *db.Deployment
+	var dbDeploymentShapes []*db.DeploymentShape
+	err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		dbDeployment, err = tx.GetDeployment(ctx, projectName, deploymentName)
+		if err != nil {
+			return fmt.Errorf("Failed loading deployment: %w", err)
+		}
+
+		dbDeploymentShapes, err = tx.GetDeploymentShapes(ctx, dbDeployment.ID)
+		if err != nil {
+			return fmt.Errorf("Failed loading deployment shapes: %w", err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	if util.IsRecursionRequest(r) {
+		deploymentShapes := make([]*api.DeploymentShape, 0, len(dbDeploymentShapes))
+		for _, dbDeploymentShape := range dbDeploymentShapes {
+			deploymentShapes = append(deploymentShapes, &dbDeploymentShape.DeploymentShape)
+		}
+
+		return response.SyncResponse(true, deploymentShapes)
+	}
+
+	deploymentShapeURLs := make([]string, 0, len(dbDeploymentShapes))
+	for _, dbDeploymentShape := range dbDeploymentShapes {
+		deploymentShapeURLs = append(deploymentShapeURLs, dbDeploymentShape.DeploymentShape.URL(version.APIVersion, dbDeployment.Name, dbDeploymentShape.Name).String())
+	}
+
+	return response.SyncResponse(true, deploymentShapeURLs)
+}
+
+// swagger:operation POST /1.0/deployments/{deploymentName}/shapes/{deploymentShapeName} deployments deployment_shapes_post
```

Comment:
Should this one be without `/{deploymentShapeName}`?
Like: `POST /1.0/deployments/{deploymentName}/shapes`

Reply:
yes it should